client/signup/utils.js

```diff
@@ -93,6 +95,25 @@ function getNextStepName( flowName, currentStepName ) {
 	return flow.steps[ indexOf( flow.steps, currentStepName ) + 1 ];
 }
 
+
+function autoFillField( config ) {
+	let { form , stepName, fieldToTake, fieldToMerge, signupProgressStore } = config;
+	if ( ! signupProgressStore ) {
+		return form;
+	}
+	let siteStepProgress = find(
+		signupProgressStore,
+		step => step.stepName === stepName
+	);
+	let fieldToMergeValue = formState.getFieldValue( form, fieldToMerge );
+	if ( fieldToMergeValue === null && siteStepProgress ) {
+		let mergeObject = {};
+		mergeObject[ fieldToMerge ] = { value: siteStepProgress[ fieldToTake ] };
+		return merge( form, mergeObject );
```

Comment:
Also unnecessary to change, but something you might find interesting nonetheless: in ES6, objects can be defined with computed keys, which can obviate `mergeObject` altogether:

```js
return merge( form, {
  [ fieldToMerge ]: { value: siteStepProgress[ fieldToTake ] }
} );
```